src/angular-app/bellows/core/offline/offline-cache.service.ts

```diff
@@ -31,10 +31,19 @@ export class OfflineCacheService {
     }));
   }
 
-  getOneFromStore(storeName: string, key: string): angular.IPromise<any> {
-    return this.$q.when(this.getStore(storeName).getItem(key).then(item => {
+  /*
+  @returns the item if found, otherwise null
+  */
+  async getOneFromStore(storeName: string, key: string): Promise<any> {
+    let item;
+    try {
+      item = await this.getStore(storeName).getItem(key);
       return OfflineCacheService.removeProjectId(item);
-    }));
+
+    } catch (err) {
+      item = null;
```

Comment:
Are we pretty okay with this pattern? If the caller's aren't prepared for null, we will get more exceptions. (Of course, it's better than it was.) Mostly wondering at a broader level if we have an established way of handling "error" cases.

Reply:
This is a good point.  I am OK with this pattern given that this is the recommended way to use the 3rd party library.  I would like to understand more about what options we have to handle these situations and what we would recommend.

Reply:
I am going to resolve this thread since it is not requesting changes and is not intended to block the PR.